misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlParser.java

```diff
@@ -148,6 +148,108 @@ private STNode parseArrayOfTables() {
+        STNode keyValuePairs = parseInlineKeyValuePairs();
+        STNode endBrace = parseCloseBrace();
+        return STNodeFactory.createInlineTableNode(openBrace, keyValuePairs, endBrace);
+    }
+
+    private STNode parseInlineKeyValuePairs() {
+        startContext(ParserRuleContext.INLINE_TABLE_LIST);
+        STToken nextToken = peek();
+
+        if (isEndOfInlineTable(nextToken)) {
+            STNode values = STNodeFactory.createEmptyNodeList();
+            endContext();
+            return values;
+        }
+        ArrayList<STNode> valuesList = new ArrayList<>();
+        
+        STNode value = parseInlineKeyValuePair();
+        valuesList.add(value);
+
+        nextToken = peek();
+        while (!(isEndOfInlineTable(nextToken))) {
+            STNode valueEnd = parseInlineTableEntryEnd();
+            if (valueEnd == null) {
+                // null marks the end of values
+                break;
+            }
+            
+            valuesList.add(valueEnd);
+            value = parseInlineKeyValuePair();
+            valuesList.add(value);
+
+            nextToken = peek();
+        }
+        endContext();
+        return STNodeFactory.createNodeList(valuesList);
+    }
+
+    private STNode parseInlineKeyValuePair() {
+        STToken nextToken = peek();
+        switch (nextToken.kind) {
+            case IDENTIFIER_LITERAL:
+            case SINGLE_QUOTE_TOKEN:
+            case DOUBLE_QUOTE_TOKEN:
+            case TRUE_KEYWORD:
+            case FALSE_KEYWORD:
+            case DECIMAL_INT_TOKEN:
+            case DECIMAL_FLOAT_TOKEN:
+                return parseKeyValue(true);
+            default:
+                recover(peek(), ParserRuleContext.INLINE_TABLE_ENTRY_START);
+                return parseInlineKeyValuePair();
+        }
+    }
+
+    private boolean isEndOfInlineTable(STToken token) {
+        return token.kind == SyntaxKind.CLOSE_BRACE_TOKEN || token.kind == EOF_TOKEN;
+    }
+
+    private STNode parseInlineTableEntryEnd() {
+        switch (peek().kind) {
+            case COMMA_TOKEN:
+                return consume();
+            case CLOSE_BRACE_TOKEN:
+                // null marks the end of values
+                return null;
+            default:
+                recover(peek(), ParserRuleContext.INLINE_TABLE_ENTRY_END);
+                return parseInlineTableEntryEnd();
+        }
+    }
+
+    private STNode parseOpenBrace() {
+        STToken token = peek();
+        if (token.kind == SyntaxKind.OPEN_BRACE_TOKEN) {
+            return consume();
+        } else {
+            recover(token, ParserRuleContext.INLINE_TABLE_START);
+            return parseOpenBrace();
+        }
+    }
+
+    private STNode parseCloseBrace() {
+        STToken token = peek();
+        if (token.kind == SyntaxKind.CLOSE_BRACE_TOKEN) {
+            return consume();
+        } else {
+            recover(token, ParserRuleContext.INLINE_TABLE_END);
+            return parseCloseBrace();
```

Comment:
These lines are not covered by a test case. Looks `inline-negative.toml` test case doesn't hit these lines.
Following sample does hit though. 
```toml
first = { first = "Tom" 
```

Reply:
It seems like parser goes to error handler in line 228 and recovers closing brace from there. Is this ok?